core/src/main/java/pt/webdetails/cda/push/WebsocketJsonQueryEndpoint.java

```diff
@@ -56,8 +53,8 @@
   @Override
   public void onMessage( String message, Consumer<String> outboundMessageConsumer ) {
     try {
-      MultivaluedMap<String, String> params = getParameters( message );
-      DoQueryParameters parameters = getDoQueryParameters( params );
+      QueryParameters.MultiValueQueryParameter<String, String> params = QueryParameters.getInstance().getParameters( message );
+      DoQueryParameters parameters = QueryParameters.getInstance().getDoQueryParameters( params );
```

Comment:
This still is a static reference ( `QueryParameters.getInstance()` ) so you didn't actually do any decoupling here. Use a getter/setter where the QueryParameters can be injected or a class attribute with construction injection. If you don't want to go through the trouble of injecting it in Spring / Blueprint you can have a default constructor that uses QueryParamenters.getInstance() as the default value for the property.